code/lib/core-server/src/utils/get-server-channel.ts

```diff
@@ -16,6 +19,27 @@ export class ServerChannel {
         });
       }
     });
+    this.webSocketServer.on('connection', (wss) => {
+      wss.on('message', (data) => {
+        try {
+          const { type, args } = JSON.parse(data.toString('utf-8'));
+          if (this.listeners) {
+            this.listeners[type].forEach((listener) => listener(args));
+          }
+        } catch (e) {
+          //
+        }
+      });
```

Comment:
We should ensure the listeners fire for events emitted on the server too (for consistency with other channels which also self-emit). In fact, can we make this a transport for `@storybook/channels`? That way we can ensure consistency.